### PR TITLE
normalize IDs for form fields in Checkout

### LINKED ISSUE #47635: Checkout fields namespace contains / but is not supported in JS querySelector
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The new [checkout additional fields](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md) uses a `/` as the namespace separator for input ids but javascript query selector throws an error when trying to query ids using `/` 

```
Failed to execute 'querySelector' on 'Document': '#additional-information-woocommerce-payments/woopay-phone-number' is not a valid selector.
```

[Here's](https://github.com/woocommerce/woocommerce/blob/a9aa14e2547b60b8970606e9418d1e763f93952c/plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php#L408C3-L414C1) where Woo checks for the namespace separator.

### Expected behavior

Not use namespace separators that make it impossible to query a field via javascript

### Actual behavior

Currently a `/` is required in the additional checkout field id but javascript throws an error when trying to query fields with `/`.

### Steps to reproduce

- Add the following snippet to a the functions.php file of your theme to add a new checkout field

```
add_action(
	'woocommerce_blocks_loaded',
	function() {
		woocommerce_blocks_register_checkout_field(
			array(
				'id'       => 'plugin-name/test-field',
				'label'    => 'Test field',
				'location' => 'additional',
				'type'     => 'checkbox',
			)
		);
	}
);
```

- Add a product to your cart
- Visit the blocks checkout page
- Open the dev tools
- Execute the following code in the console:

```
document.querySelector('#plugin-name/test-field');
```
- An error will be thrown

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by senadir**
With additional fields, we use the whole field key as a form ID, which will always have a `/` separator, the issue here is that html ID don't support `/` and using `querySelector` will throw an error. For that, we normalize the `/` to `-`.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/47635

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following snippet somewhere (or using a snippet plugin):
```php
add_action(
	'woocommerce_init',
	function () {
		woocommerce_register_additional_checkout_field(
			array(
				'id'       => 'plugin-name/test-field',
				'label'    => 'Test field',
				'location' => 'additional',
				'type'     => 'checkbox',
			)
		);
	}
);
```
2. Add an item to cart and go to Checkout.
3. Using console, run the following code: `document.querySelector('#additional-information-plugin-name-test-field')`.
4. There should be no errors.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Replace forward slashes in additional fields IDs with hyphens. 

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx [hunk 1]
```diff
@@ -9,14 +9,14 @@ import { __, sprintf } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
-import { AddressFieldProps } from './types';
+import { AddressLineFieldProps } from './types';
 
-const Address2Field = < T extends AddressFormValues | ContactFormValues >( {
+const AddressLine2Field = < T extends AddressFormValues | ContactFormValues >( {
```
Comment (senadir) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:14]: I normalized `AddressField`, `AddressFields`, and `Address2Field` usage to AddressLine. cc @nielslange

Comment (senadir) [reply plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:14]: This is to reduce confusion with the address form and fields.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-fields.tsx [hunk 2]
```diff
@@ -7,22 +7,22 @@ import { AddressFormValues, ContactFormValues } from '@woocommerce/settings';
...
-import Address2Field from './address-2-field';
-import { AddressFieldsProps } from './types';
+import AddressLine2Field from './address-line-2-field';
+import { AddressLineFieldsProps } from './types';
 import { createFieldProps } from './utils';
 
-const AddressFields = < T extends AddressFormValues | ContactFormValues >( {
-	id,
+const AddressLineFields = < T extends AddressFormValues | ContactFormValues >( {
+	formId,
```
Comment (senadir) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-fields.tsx:15]: This is not the field id, but the form id.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/utils.ts [hunk 3]
```diff
@@ -25,19 +25,19 @@ export interface FieldProps {
...
-	id: `${ fieldId }-${ address?.key }`,
-	errorId: `${ fieldAddressType }_${ address?.key }`,
-	label: address?.required ? address?.label : address?.optionalLabel,
-	autoCapitalize: address?.autocapitalize,
-	autoComplete: address?.autocomplete,
-	errorMessage: address?.errorMessage,
-	required: address?.required,
-	className: `wc-block-components-address-form__${ address?.key }`,
-	...address?.attributes,
+	id: `${ formId }-${ field?.key }`.replaceAll( '/', '-' ), // Replace all slashes with hyphens to avoid invalid HTML ID.
```
Comment (senadir) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/utils.ts:32]: Actual fix here.

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md [hunk 4]
```diff
@@ -347,7 +347,7 @@ This example demonstrates rendering a text field in the address section:
 
 ```php
 add_action(
-	'woocommerce_blocks_loaded',
+	'woocommerce_init',
```
Comment (senadir) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md:350]: woocommerce_init works for both themes and plugins and snippet plugins.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx (hunk 1 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:14]: Sounds good, @senadir. Thanks for improving this.

**Review by opr — APPROVED** [review]
Looks good, thanks!

**Review by opr — APPROVED** [review]
Looks good, thanks!

**Pull request merged**